api/app/tests/apps/api/v1/test_public_map_signals_endpoint.py

```diff
@@ -0,0 +1,56 @@
+import os
+
+from django.test import override_settings
+
+from signals.apps.api.v1.urls import SignalsRouterVersion1
+from signals.apps.api.v1.views import PublicSignalListViewSet
+from signals.apps.signals.factories import SignalFactoryValidLocation
+from tests.test import SignalsBaseApiTestCase
+
+THIS_DIR = os.path.dirname(__file__)
+
+
+class NameSpace():
+    pass
+
+
+# we need to simulate a new route or unit test, urls.py is evaluated once, and cannot be unit tested with settings
+# we set a custom route for the unit test only
+extra_router = SignalsRouterVersion1()
+extra_router.register(r'public/map-signals', PublicSignalListViewSet, basename='public-list-signals')
+test_urlconf = NameSpace()
+test_urlconf.urlpatterns = extra_router.urls
+
+
+@override_settings(ROOT_URLCONF=test_urlconf)
+class TestMapSignalEndpoints(SignalsBaseApiTestCase):
+    def setUp(self):
+        self.endpoint_url = '/public/map-signals/'
+        self.signal1 = SignalFactoryValidLocation.create()
+        self.signal2 = SignalFactoryValidLocation.create()
+        super(TestMapSignalEndpoints, self).setUp()
+
+    def test_map_signals_list(self):
+        response = self.client.get(self.endpoint_url)
+        self.assertEqual(response.status_code, 200)
+        data = response.json()
+        self.assertEqual(len(data['features']), 2)
+        datadict = {k[1]['properties']['id']: k[1] for k in enumerate(data['features'])}
```

Comment:
No need to use `enumerate` here. You are not interested in the index within the list of features just the feature itself. The following is shorter and more readable:

`datadict = {feature['properties']['id']: feature for feature in data['features']}`

Reply:
done